lib/index.js

```diff
@@ -190,8 +192,11 @@ exports.register = function (server, options, next) {
 
     server.ext('onPreResponse', function (request, reply) {
 
+        var skipCacheStorage = !settings.errorOnCacheNotReady && !cache.isReady();
+
         if (!request.session ||
-            (!request.session._isModified && !request.session._isLazy)) {
+            skipCacheStorage ||
```

Comment:
Good point.  To that point, the check on preResponse is a tad too early too.  I will move that check to the line that checks if the cookie returned session data and add a test for that.